cwms_radar_api/src/main/java/cwms/radar/data/dao/TimeSeriesGroupDao.java

```diff
@@ -27,70 +37,141 @@ public TimeSeriesGroupDao(DSLContext dsl)
 
 	public List<TimeSeriesGroup> getTimeSeriesGroups(String officeId)
 	{
-		AV_TS_CAT_GRP table = AV_TS_CAT_GRP.AV_TS_CAT_GRP;
+		Condition whereCond = null;
+		if(officeId != null)
+		{
+			whereCond = AV_TS_GRP_ASSGN.AV_TS_GRP_ASSGN.DB_OFFICE_ID.eq(officeId);
+		}
 
-		org.jooq.TableField [] columns = new TableField[]{
-				table.CAT_DB_OFFICE_ID, table.TS_CATEGORY_ID,
-				table.TS_CATEGORY_DESC, table.GRP_DB_OFFICE_ID, table.TS_GROUP_ID,
-				table.TS_GROUP_DESC, table.SHARED_TS_ALIAS_ID, table.SHARED_REF_TS_ID
-		};
+		return getTimeSeriesGroupsWhere(whereCond);
+	}
 
-		SelectJoinStep<Record> step
-				= dsl.selectDistinct(columns)
-				.from(table);
+	@NotNull
+	private List<TimeSeriesGroup> getTimeSeriesGroupsWhere(Condition whereCond)
+	{
+		List<TimeSeriesGroup> retval = new ArrayList<>();
+		AV_TS_CAT_GRP catGrp = AV_TS_CAT_GRP.AV_TS_CAT_GRP;
+		AV_TS_GRP_ASSGN grpAssgn = AV_TS_GRP_ASSGN.AV_TS_GRP_ASSGN;
 
-		Select select = step;
+		final RecordMapper<Record, Pair<TimeSeriesGroup, AssignedTimeSeries>> mapper = record15 -> {
+			TimeSeriesGroup group = buildTimeSeriesGroup(record15);
+			AssignedTimeSeries loc = buildAssignedTimeSeries(record15);
 
-		if ( officeId != null && !officeId.isEmpty())
+			return new Pair<>(group, loc);
+		};
+
+		SelectOnConditionStep<Record16<String, String, String, String, String, String, String, String, String, String, String, String, BigDecimal, BigDecimal, String, String>>
```

Comment:
FYI, if you're always using the named based getters later you can just do something like `SelectOnConditionStep<?>`

Not a huge deal but I imagine you like typing these out as much as I do.

Reply:
That makes sense.